lib/CLI/Parser.hs

```diff
@@ -505,29 +498,11 @@ expectedQualifiedPathFormat = Pretty.vsep
 
 expectedFilterFormat :: Pretty.Doc
 expectedFilterFormat = Pretty.vsep
-  [ "Expected format is: 'name~<substring>' or `date<op><date>`."
-  , "<op> can be '<=', '>=', '<', '>', or '='."
-  , "<date> can be 'YYYY', 'YYYY-MM', 'YYYY-MM-DD', or 'YYYY-MM-DD HH:MM:SS'."
-  , "Examples: 'name~vault', 'date<2020-02'."
-  ]
-
-readFilterField :: ReadM (FieldName, FilterField)
-readFilterField = do
-  eitherReader \input ->
-    P.parse (parseFilterField <* P.eof) "" (T.pack input) & first \err -> unlines
-      [ "Invalid filter-field format: " <> show input <> "."
-      , show expectedFilterFieldFormat
-      , ""
-      , "Parser error:"
-      , P.errorBundlePretty err
-      ]
-
-expectedFilterFieldFormat :: Pretty.Doc
-expectedFilterFieldFormat = Pretty.vsep
-  [ "Expected format is: '<fieldname>:contents~<substring>' or `<fieldname>:date<op><date>`."
+  [ "Expected format is: 'name~<substring>' or 'date<op><date>',"
+  , "'<fieldname>:contents~<substring>' or '<fieldname>:date<op><date>'."
```

Comment:
Hmm, now that we support 4 different formats, maybe displaying them in a bulleted list would look better? And a short description of what which format does.

See the example in `expectedSortFormat`.

Reply:
Done. But I noticed that the error messages when parsing `filter` option differ from `sort`.
When using `--sort`, we cannot see any parsing exceptions in raw form, e.g.:
```
>>> coffer find --sort name:
option --sort: Invalid direction: ''.
Choose one of: 'asc', 'desc'.


Usage: coffer find [PATH] [TEXT] [--sort SORT] [--filter FILTER]
  Find and list entries, optionally filtering
```

But in case of `--filter` the raw parse exceptions appear (and it was before this pull request), e.g.:
```
>>> coffer find --filter FILTER
...
Parser error:
1:7:
  |
1 | FILTER
  |       ^
unexpected end of input
expecting ':' or fieldname
...
```
I think that it would be nice if we chose one style of reporting errors.

Reply:
Ah, that seems to be because we're using megaparsec to parse filters (which have a more complex syntax) and `Text.splitOn ":"` to parse sorts (which have a simpler syntax).

Yeah, we could use megaparsec for both, for consistency. Please create a separate issue for that.